File: json-schema-validation/src/test/java/io/micronaut/jsonschema/validation/RWBlackbird.java
package io.micronaut.jsonschema.validation;

import io.micronaut.core.annotation.Introspected;
import io.micronaut.jsonschema.JsonSchema;

/**
 * A red-winged blackbird.
 *
 * @param name The name
 * @param wingSpan The wing span of the bird
 */
// tag::clazz[]
@JsonSchema(
    title = "RedWingedBlackbird", // <1>
    description = "A species of blackbird with red wings",
    uri = "/red-winged-blackbird" // <2>
)
@Introspected
public record RWBlackbird(
    String name,
    double wingSpan
) {
}
// end::clazz[]

File: test-suite-python/src/test/python/micronaut/jsonschema/test/RWBlackbird.py
from dataclasses import dataclass

from micronaut.jsonschema import JsonSchema
from micronaut.serde.annotation import Serdeable


# tag::clazz[]
@JsonSchema(
    title="RedWingedBlackbird",  # <1>
    description="A species of blackbird with red wings",
    uri="/red-winged-blackbird",  # <2>
)
@Serdeable
@dataclass
class RWBlackbird:
    """A red-winged blackbird."""

    name: str | None = None
    """The name"""

    wingSpan: float | None = None
    """The wing span of the bird"""
# end::clazz[]
